### PR TITLE
Source Stripe: remove duplicate streams

### PR DESCRIPTION
**Pull request by clnoll**
## What
Fixes the root cause of https://github.com/airbytehq/oncall/issues/2578, in which syncs were failing due to duplicate streams.

## How
Removes duplicate streams.

This should be tested using a version override in LaunchDarkly before we unpin the cloud version.

### CODE REVIEW

File: airbyte-integrations/connectors/source-stripe/metadata.yaml [hunk 1]
```diff
@@ -5,7 +5,7 @@ data:
   connectorSubtype: api
   connectorType: source
   definitionId: e094cb9a-26de-4645-8761-65c0c425d1de
-  dockerImageTag: 3.14.0
+  dockerImageTag: 3.15.0
```
Comment (erohmensing) [inline airbyte-integrations/connectors/source-stripe/metadata.yaml:8]: please remove the version pin on L17!

**Comment by octavia-squidington-iii** [comment]
##  source-stripe test report (commit [`c3a22ae3e6`](https://github.com/airbytehq/airbyte/pull/28709/commits/c3a22ae3e67ad4218c6520657a4223c4e7d0f133)) - ❌

⏲️  Total pipeline duration: 01mn20s 

| Step                                                                 | Result   |
|:---------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-stripe/metadata.yaml | ✅        |
| Connector version semver check                                       | ✅        |
| QA checks                                                            | ❌        |
| Code format checks                                                   | ✅        |
| Connector package install                                            | ✅        |
| Build source-stripe docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                           | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/source-stripe-remove-duplicate-streams/1690323459/c3a22ae3e67ad4218c6520657a4223c4e7d0f133/source-stripe/3.15.0/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-stripe test
```

**Review by erohmensing — APPROVED** [review]
LGTM, would be great if you could test it on a local oss deployment!  

Alternatively we could not remove the pin quite yet, and put our internal workspace on cloud up to the 3.15.0 version to make sure it works 

Some checks will fail until you add a changelog, but I assume you can handle that

**Comment by erohmensing** [comment]
We also need to fix the failing unit test, I guess - this should have been a giveaway to the problem: https://github.com/airbytehq/airbyte/commit/66938e1e69fbf22bc343dea360937b1fe58aebfb#diff-aa55e9435ad84f183d0e3812bcafb15011e2614a57a290aef1b447618ba27232R45

**Comment by octavia-squidington-iii** [comment]
##  source-stripe test report (commit [`ff15c7db63`](https://github.com/airbytehq/airbyte/pull/28709/commits/ff15c7db631add29185510643d170b27cb2520c3)) - ❌

⏲️  Total pipeline duration: 01mn27s 

| Step                                                                 | Result   |
|:---------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-stripe/metadata.yaml | ✅        |
| Connector version semver check                                       | ✅        |
| QA checks                                                            | ✅        |
| Code format checks                                                   | ✅        |
| Connector package install                                            | ✅        |
| Build source-stripe docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                           | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/source-stripe-remove-duplicate-streams/1690324320/ff15c7db631add29185510643d170b27cb2520c3/source-stripe/3.15.0/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-stripe test
```

File: airbyte-integrations/connectors/source-stripe/metadata.yaml (hunk 1 above)
Comment (clnoll) [reply airbyte-integrations/connectors/source-stripe/metadata.yaml:8]: I was thinking I'd do that in a separate PR, after testing this on our internal workspace using a LD version override.

Comment (erohmensing) [reply airbyte-integrations/connectors/source-stripe/metadata.yaml:8]: Got it, works for me!

**Comment by octavia-squidington-iii** [comment]
##  source-stripe test report (commit [`23da2d08fa`](https://github.com/airbytehq/airbyte/pull/28709/commits/23da2d08fa0cf953644ed8f5ba16ef96794fff00)) - ✅

⏲️  Total pipeline duration: 11mn23s 

| Step                                                                 | Result   |
|:---------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-stripe/metadata.yaml | ✅        |
| Connector version semver check                                       | ✅        |
| QA checks                                                            | ✅        |
| Code format checks                                                   | ✅        |
| Connector package install                                            | ✅        |
| Build source-stripe docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                           | ✅        |
| Acceptance tests                                                     | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/source-stripe-remove-duplicate-streams/1690324632/23da2d08fa0cf953644ed8f5ba16ef96794fff00/source-stripe/3.15.0/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-stripe test
```

**Comment by clnoll** [comment]
Confirming that the sync succeeded locally. Will also test this version against our internal connector before unpinning the version.

**Pull request merged**